tests/sles4sap/qesapdeployment/deploy.pm

```diff
@@ -42,6 +46,32 @@ sub run {
 
     }
 }
+sub clean_up {
+    my ($self) = @_;
+    my @cmds_list;
+    # Only run the Ansible deregister if the inventory is present
+    push(@cmds_list, 'ansible') if (!script_run 'test -f ' . qesap_get_inventory(get_required_var('PUBLIC_CLOUD_PROVIDER')));
+
+    # Terraform destroy can be executed in any case
+    push(@cmds_list, 'terraform');
+
+    for my $command (@cmds_list) {
+        record_info('Cleanup', "Executing $command cleanup");
+        my @clean_up_cmd_rc = qesap_execute(verbose => '--verbose', cmd => $command, cmd_options => '-d', timeout => 1200);
+        if ($clean_up_cmd_rc[0] == 0) {
+            diag(ucfirst($command) . " cleanup attempt #  PASSED.");
+            record_info("Clean $command", ucfirst($command) . ' cleanup PASSED.');
+            last;
+        }
+        else {
+            diag(ucfirst($command) . " cleanup attempt #  FAILED.");
+            sleep 10;
+            record_info('Cleanup FAILED', "Cleanup $command FAILED", result => 'fail') if $clean_up_cmd_rc[0];
```

Comment:
the `if` at the end of this line is not needed. We are already within a if/else codeblock controlled by `$clean_up_cmd_rc[`

Reply:
Removed if conditions in else block